src/core/expr/fexpr_cut.cc

```diff
@@ -67,12 +73,31 @@ class FExpr_Cut : public FExpr_Func {
         throw NotImplError() << "cut() cannot be used in a groupby context";
       }
 
-      int32_t nbins_default = 10;
       Workframe wf = arg_->evaluate_n(ctx);
+
+      bool defined_bins = !py_bins_.is_none();
+      bool defined_nbins = !py_nbins_.is_none();
+
+      if (defined_bins && defined_nbins) {
+        throw ValueError() << "`bins` and `nbins` cannot be both set at the same time";
+      }
```

Comment:
Validation of parameters should generally be done before the evaluation stage. This check, for example, can be performed as early as in `pyfn_cut`